cmd/admin-handlers.go

```diff
@@ -2465,44 +2472,268 @@ func fetchLambdaInfo() []map[string][]madmin.TargetIDStatus {
-	switch {
-	case err != nil:
-		kmsStat.Decrypt = fmt.Sprintf("Decryption failed: %v", err)
-	case subtle.ConstantTimeCompare(key.Plaintext, decryptedKey) != 1:
-		kmsStat.Decrypt = "Decryption failed: decrypted key does not match generated key"
-	default:
-		kmsStat.Decrypt = "success"
+	var response KESStatus
+	if err = json.NewDecoder(mem.LimitReader(resp.Body, 1*mem.MiB)).Decode(&response); err != nil {
+		return madmin.KMS{}, err
+	}
+	return madmin.KMS{Status: string(madmin.ItemOnline), Endpoint: endpoint, Version: response.Version}, nil
+}
+
+// generateKey generates a sample key
+func generateKey(keyID string, cryptoCtx kms.Context, client http.Client, endpoint string) (kms.DEK, error) {
+	ctxBytes, err := cryptoCtx.MarshalText()
+	if err != nil {
+		return kms.DEK{}, err
+	}
+
+	type Request struct {
+		Context []byte `json:"context,omitempty"` // A context is optional
+	}
+	type Response struct {
+		Plaintext  []byte `json:"plaintext"`
+		Ciphertext []byte `json:"ciphertext"`
+	}
+
+	body, err := json.Marshal(Request{
+		Context: ctxBytes,
+	})
+	if err != nil {
+		return kms.DEK{}, err
+	}
+
+	request, err := http.NewRequestWithContext(context.Background(), http.MethodPost, endpoint+"/v1/key/generate", bytes.NewReader(body))
+	request.Header.Set("Content-Type", "application/json")
+	if err != nil {
+		return kms.DEK{}, err
+	}
+	resp, err := client.Do(request)
+	if err != nil {
+		return kms.DEK{}, err
+	}
+	defer resp.Body.Close()
+
+	var response Response
+	if err = json.NewDecoder(mem.LimitReader(resp.Body, 1*mem.MiB)).Decode(&response); err != nil {
+		return kms.DEK{}, err
+	}
+
+	return kms.DEK{
+		KeyID:      keyID,
+		Plaintext:  response.Plaintext,
+		Ciphertext: response.Ciphertext,
+	}, nil
+}
+
+// decryptKey decrypts a given cipher
+func decryptKey(keyID string, ciphertext []byte, ctx kms.Context, client http.Client, endpoint string) ([]byte, error) {
+	ctxBytes, err := ctx.MarshalText()
+	if err != nil {
+		return nil, err
+	}
+
+	type Request struct {
+		Ciphertext []byte `json:"ciphertext"`
+		Context    []byte `json:"context,omitempty"` // A context is optional
+	}
+	type Response struct {
+		Plaintext []byte `json:"plaintext"`
+	}
+	body, err := json.Marshal(Request{
+		Ciphertext: ciphertext,
+		Context:    ctxBytes,
+	})
+	if err != nil {
+		return nil, err
+	}
+	request, err := http.NewRequestWithContext(context.Background(), http.MethodPost, endpoint+"/v1/key/decrypt", bytes.NewReader(body))
+	request.Header.Set("Content-Type", "application/json")
+	if err != nil {
+		return nil, err
+	}
+	resp, err := client.Do(request)
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+
+	var response Response
+	if err = json.NewDecoder(mem.LimitReader(resp.Body, 1*mem.MiB)).Decode(&response); err != nil {
+		return nil, err
+	}
+	return response.Plaintext, nil
+}
+
+// getKesClient returns an http client for KES
+func getKesClient() http.Client {
```

Comment:
You can actually just return a `kes.Client` or use the `internal/kms.NewWithConfig` function.